pdns/protozero-trace.cc

```diff
@@ -557,25 +558,44 @@ void extractOTraceIDs(const EDNSOptionViewMap& map, pdns::trace::InitialSpanInfo
   // traceid gets set from edns options (if available and well-formed), otherwise random
   // parent_span_id gets set from edns options (if available and well-formed, otherwise it remains cleared (no parent))
   // span_id gets inited randomly
-  bool traceidset = false;
+  auto traceidset = extractOTraceIDs(map, EDNSOptionCode::OTTRACEIDS, span.trace_id, span.parent_span_id);
```

Comment:
I see a conflict here with  #16362, it doe does longer set the ID to random if not found. I do that later there, as I might not be using the trace.